### ISSUE TITLE
optimize `Model.save` performance

### ISSUE THREAD

**Issue by huangwei1024** [josiahcarlson/rom#54]
```python
class Model(six.with_metaclass(_ModelMetaclass, object)):
def save(self, full=False):
        '''
        Saves the current entity to Redis. Will only save changed data by
        default, but you can force a full save by passing ``full=True``.
        '''
        new = self.to_dict()
        ret, rnew = self._apply_changes(self._last, new, full or self._new)
        self._new = False
        self._last = rnew
        #last = {}
        #cols = self._columns
        #for attr, data in new.items():
        #    last[attr] = cols[attr]._to_redis(data) if data is not None else None
        #self._last = last
```

`Model._apply_changes` has data write to redis, why not use it?

**Comment by josiahcarlson** [comment]
Either I don't understand what you mean by "optimize", or you are mistaken as to what amount of performance improvement is possible by moving code.

`Model._apply_changes()` is used by both `Model.save()` and `Model.delete()`, so unless I want to copy/paste the same code and keep it updated in two places (a very bad idea when it comes to maintenance), the use of `Model._apply_changes()` is the right answer.

So... what are you actually proposing, and what optimization are you claiming?

**Comment by huangwei1024** [comment]
Sorry for my poor english.
I mean this code could be simplify.

        # Now explicitly encode data for the _last attribute to make re-saving
        # work correctly in all cases.
        last = {}
        cols = self._columns
        for attr, data in new.items():
            last[attr] = cols[attr]._to_redis(data) if data is not None else None

        self._last = last

it reduce `cols[attr]._to_redis(data)` be called again. some changed columns call `_to_redis` in `_apply_changes`, I think its could be reuse.
In current code, the all columns will be invoke `_to_redis` even if the columns not be changed.
the understand of mine was correct?

**Comment by thedrow** [comment]
@huangwei1024 I'm having a really hard time understanding you.
Could you please provide the exact code you want to change in a Pull Request and explain what's better about it?

**Comment by huangwei1024** [comment]
I push a request, plz check it.

**Comment by josiahcarlson** [comment]
This should be addressed in the most recent commit, which has been released as 0.29.3 in the cheeseshop and is tagged as such here. Closing as fixed.
